lib/fibonacci.rb

```diff
@@ -8,5 +8,20 @@
 # Time complexity: ?
 # Space complexity: ?
```

Comment:
Complexity analysis?

Reply:
oops I forgot to push this update. I came up with the following:
# Time complexity: O(n), n is the number of computations this calculation needs to run to compute the nth fibonacci number
# Space complexity: O(1), space complexity is constant because this method only require a storage for 3 variables (p,p1,p2) regardless of number of operations